rust/lance/src/index/cache.rs

```diff
@@ -19,17 +19,37 @@ use moka::sync::{Cache, ConcurrentCacheExt};
 
 use super::vector::VectorIndex;
 
+use std::sync::atomic::{AtomicU64, Ordering};
+
+#[derive(Debug, Default)]
+pub struct CacheStats {
+    hits: AtomicU64,
+    misses: AtomicU64,
+}
+
+impl CacheStats {
+    pub fn record_hit(&self) {
+        self.hits.fetch_add(1, Ordering::Relaxed);
```

Comment:
With relaxed ordering here you will get inaccurate counts, but I suppose that isn't that much of a problem for this sort of counter.

Reply:
Yeah, I'd prefer not to risk index latency for potential accuracy here.